cmd/server-main.go

```diff
@@ -223,8 +226,57 @@ func serverMain(ctx *cli.Context) {
 	handleSignals()
 }
 
+// Lists all your local mount points belonging to a given prefix.
+func listMountsPrefix(prefix string) ([]mountinfo.MountPoint, error) {
+	mounts, err := mountinfo.MountInfo()
+	if err != nil {
+		// Ignore if mountinfo thinks /proc/mounts is missing
+		// or minio has no permission to read it. In such cases
+		// we just treat it as success and hopefully report
+		// cross-device issues if any at a later stage.
+		if os.IsNotExist(err) || os.IsPermission(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	var newMounts []mountinfo.MountPoint
```

Comment:
@harshavardhana, you need to add the following code:
```
        prefix, err = filepath.Abs(prefix)
        if err != nil {
                return nil, err
        }
        prefix += string(filepath.Separator)
```

Using Abs() is more robust when comparing with mounted paths.

Besides, you need to add a separator because otherwise, the user cannot run 'minio server /tmp/fs' if there is another mounted directory which starts by fs, like /tmp/fsmount.